modules/mauth-authenticator-http4s/src/main/scala/com/mdsol/mauth/http4s/MauthPublicKeyProvider.scala

```diff
@@ -23,23 +22,39 @@ import com.mdsol.mauth.http4s.MauthPublicKeyProvider.SecurityToken
 import io.circe.{Decoder, HCursor}
 import org.http4s.circe.CirceEntityDecoder._
 import org.typelevel.log4cats.Logger
+import cats.effect.implicits._
 
-class MauthPublicKeyProvider[F[_]: Concurrent: Logger](configuration: AuthenticatorConfiguration, signer: MAuthRequestSigner, val client: Client[F])(implicit
-  val cache: Cache[F, String, Option[PublicKey]]
+class MauthPublicKeyProvider[F[_]: Async: Logger](configuration: AuthenticatorConfiguration, signer: MAuthRequestSigner, val client: Client[F])(implicit
+  val cache: Cache[F, UUID, F[Option[PublicKey]]]
 ) extends ClientPublicKeyProvider[F] {
 
   /** Returns the associated public key for a given application UUID.
     *
     * @param appUUID , UUID of the application for which we want to retrieve its public key.
     * @return { @link PublicKey} registered in MAuth for the application with given appUUID.
     */
-  override def getPublicKey(appUUID: UUID): F[Option[PublicKey]] = memoizeF(Some(configuration.getTimeToLive.seconds)) {
-    val uri = new URI(configuration.getBaseUrl + getRequestUrlPath(appUUID))
-    val signedRequest = signer.signRequest(UnsignedRequest.noBody("GET", uri, headers = Map.empty))
-    signedRequest
-      .toHttp4sRequest[F]
-      .flatMap(req => client.run(req).use(retrievePublicKey))
-  }
+  override def getPublicKey(appUUID: UUID): F[Option[PublicKey]] = cache
```

Comment:
How I would love that this was an actual `@newtype case class AppUUID(uuid: UUID)` instance, instead of a random `UUID` :-(